crates/sui-types/src/base_types.rs

```diff
@@ -306,6 +300,12 @@ pub struct ObjectDigest(
     pub [u8; OBJECT_DIGEST_LENGTH],
 ); // We use SHA3-256 hence 32 bytes here
 
+impl IntoPoint for ObjectDigest {
+    fn into_point(&self) -> RistrettoPoint {
+        RistrettoPoint::hash_from_bytes::<Sha512>(&self.0)
```

Comment:
we shouldn't need to re-hash a hash, should we?

Reply:
I did take a look at that too, but decided to leave it as it is for now.
The primary reason is that RistrettoPoint uses 512-bit array, while our digests are 256-bit.
So if we were to use object digest directly we will need to decide how to transform the array. I don't know RistrettoPoint enough to be sure that the transformation won't somehow affect its security properties.